pkg/cmd/release/create/create.go

```diff
@@ -409,20 +598,38 @@ func selectPackageOverrides(octopus *octopusApiClient.Client, ask question.Asker
 	return "", nil
 }
 
-func selectProject(octopus *octopusApiClient.Client, ask question.Asker) (*projects.Project, error) {
-
+func selectProject(octopus *octopusApiClient.Client, ask question.Asker, spinner factory.Spinner) (*projects.Project, error) {
+	spinner.Start()
 	existingProjects, err := octopus.Projects.GetAll()
+	spinner.Stop()
 	if err != nil {
 		return nil, err
 	}
 
 	return question.SelectMap(ask, "Select the project in which the release will be created", existingProjects, func(p *projects.Project) string {
-		for _, v := range existingProjects {
-			if p.Name == v.Name {
-				return v.Name
-			}
-		}
+		return p.Name
+	})
+}
 
-		return ""
+func selectGitReference(octopus *octopusApiClient.Client, ask question.Asker, spinner factory.Spinner, project *projects.Project) (*projects.GitReference, error) {
+	spinner.Start()
+	branches, err := octopus.Projects.GetGitBranches(project)
```

Comment:
This is the dangerous part here; there could potentially be hundreds or even thousands of branches and tags in a large repo. This just dumbly loads them all